File: Week_04/greedy/55_jump_game.java
class Solution {
    public boolean canJump(int[] nums) {
        int furthest = 0;
        for(int i=0; i< nums.length; i++) {
            if( i > furthest) return false;

            furthest = Math.max(furthest, i + nums[i]);
        }

        return true;
    }

}

File: Week_01/python/55_jump_game.py
from typing import List


class Solution:
    def canJump(self, nums: List[int]) -> bool:
        furthest = 0

        for i, num in enumerate(nums):
            if furthest >= i:
                furthest = max(i + num, furthest)

        return furthest>= len(nums)-1
